### PR TITLE
feat(owlbot-java): owlbot-java can now migrate a repo from autosynth to owlbot

### PR DESCRIPTION
**Pull request by chingor13**
Uses python's `ast` module to parse a `synth.py` and find critical arguments to codegen (service name, destination name, path to protos, etc). If possible, write a simplified `owlbot.py` and configure a `.github/.OwlBot.yaml` from the detected configuration.

### CODE REVIEW

File: docker/owlbot/java/src/convert-synth.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import ast
+import os
+import pathlib
+from typing import Union
+
+import click
+from jinja2 import Environment, FileSystemLoader
+
+
+root_directory = pathlib.Path(
```
Comment (SurferJeffAtGoogle) [inline docker/owlbot/java/src/convert-synth.py:24]: Please add comments and doc strings to this whole file.

Comment (chingor13) [reply docker/owlbot/java/src/convert-synth.py:32]: Done.

**Comment by chingor13** [comment]
Can probably leave the migration script separate from the owlbot image

**Comment by chingor13** [comment]
This is moved to our internal repo bootstrapping tool for Java.

**Pull request closed without merging**